### PR TITLE
Bug fixes in Settings form

### PR DESCRIPTION
**Pull request by ananelson**
… existing credentials when client info changes.

Checkbox bug fix needed for new features involving boolean settings.

Credential invalidation code bug fix - although FYI this doesn't fix the issue I'm having where changing any setting causes credentials to be invalidated.

### CODE REVIEW

File: CRM/Civiquickbooks/Form/Settings.php [hunk 1]
```diff
@@ -139,11 +139,25 @@ public function getFormSettings() {
   public function saveSettings() {
     $settings = $this->getFormSettings();
     $values = array_intersect_key($this->_submittedValues, $settings);
-    $previousValues = CRM_Quickbooks_APIHelper::getQuickBooksCredentials();
+
+    // Fix for unsetting a checkbox. When setting a checkbox,
```
Comment (agileware-fj) [inline CRM/Civiquickbooks/Form/Settings.php:143]: This section that adds the boolean settings fix looks good, although it looks as though it actually belongs to a different PR

File: CRM/Civiquickbooks/Form/Settings.php [hunk 2]
```diff
@@ -139,11 +139,25 @@ public function getFormSettings() {
...
+    // attribute should be interpreted as setting its value to 0.
+    foreach ($settings as $setting_name => $setting_info) {
+        if ($setting_info['type'] == "Boolean" && !array_key_exists($setting_name, $values)) {
+            $values[$setting_name] = 0;
+        }
+    }
 
     civicrm_api3('setting', 'create', $values);
 
-    if ($previousValues['clientID'] != 'quickbooks_consumer_key' || $previousValues['clientSecret'] != 'quickbooks_shared_secret') {
```
Comment (agileware-fj) [inline CRM/Civiquickbooks/Form/Settings.php:146]: Need to confirm the issue & do run testing on this section before we can merge.

Comment (ananelson) [reply CRM/Civiquickbooks/Form/Settings.php:146]: ok sounds like I should leave this one out for now? I have a bunch of other more relevant code I'd love to PR instead

**Comment by ananelson** [comment]
ok - just the one bug fix for now
